### PR TITLE
History of Nodejs

### PR DESCRIPTION
**Pull request by Jethro-magaji**
This is the rewritten and corrected article on the history of nodejs based on the previous recommendations.

### CODE REVIEW

**Comment by hectorkambow** [comment]
Duplicate of this [PR](https://github.com/section-io/engineering-education/pull/232)

**Comment by Jethro-magaji** [comment]
Hello,
Please, when you say " Duplicate of this PR" what do you mean ?

**Comment by hectorkambow** [comment]
It means there are two PR with the same content - its for us to keep track. @Jethro-magaji

File: articles/history-of-nodejs/history-of-nodejs.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    - Fast
+    - No Buffering
+    - Consistent source code
+    - Massive Library support
+
+### The Early Beginnings
+Believe it or not, Node.js is only twelve years old. In comparison, JavaScript is 26 years old and the Web is 32 years old.
+[Ryan Dahl wrote Node.js](https://en.wikipedia.org/wiki/Node.js) about thirteen ago, in 2009. At first Node.js supported only Linux and Mac OS X. Dahl led the development and maintenance and later on it was [sponsored by Joyent](https://en.wikipedia.org/wiki/Node.js).
+The most popular web server of the time (Apache) was criticized back in 2009 due to its limited possibilities, because the Apache HTTP Server had to handle various connections concurrently(up to 10,000 or more) which were beyond its capabilities.
+The way of creating code was commonly through sequential programming, when there was either a blocked code in the entire process or an implied multiple execution stacks in cases of simultaneous connections.
```
Comment (hectorkambow) [inline articles/history-of-nodejs/history-of-nodejs.md:46]: line 46 & 47 need some rewording - a bit difficult to read through or understand.  I think its trying to explain why apache has difficulties or was not the best option. Read lines above it to get a better sense of the flow - of my suggested edits so far.

File: articles/history-of-nodejs/history-of-nodejs.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+### The Evolution
+![Nodejs image evolution](https://drive.google.com/uc?export=view&id=1e91n-R0-W4S6UOffTNTnBdqyJKNqzOXN)
+[source](https://nodejs.dev/learn/a-brief-history-of-nodejs)
+
+As many browsers competed to offer users the best performance, JavaScript engines also became considerably better. Because of this major browsers worked hard to find ways to make JavaScript run quicker and ways to offer better support for it.
+It wasn't only luck that made Node.js as popular as it is today, but one could say it was built at the right place and at the right time. It introduces a lot of approaches for JavaScript server-side development and [innovative thinking](https://nodejs.dev/learn/a-brief-history-of-nodejs) that has helped many developers.
+
+Node.js has thousands of open-source libraries and the npm website is where most of them are hosted. There are a lot of developer events and conferences that support the Node.js community, these includes Node Interactive, Node Summit and NodeConf as well as other regional events.
```
Comment (hectorkambow) [inline articles/history-of-nodejs/history-of-nodejs.md:56]: Id suggest including the npm website to help any developers reading along - that may need it

**Comment by Jethro-magaji** [comment]
I just made a commit based on the review suggestions.

File: articles/history-of-nodejs/history-of-nodejs.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    alt:
+---
+Today we are going to talk about a revolutionary technology that is being used to build great web applications, it is called  *Node.js*.
+You may have probably heard of it or worked with it (or have been curious about it in the past), either way you are welcome to join us as we further explore the History of Node.js in this article. We are going to talk about the history, some of the basics, and a few use case examples that use this revolutionary technology.
+<!--more-->
+![Nodejs image](/engineering-education/history-of-nodejs/nodejs-use-cases-cover-image.png)
+[source](https://www.simform.com/nodejs-use-case/)
+
+### What is Node.js?
+![Nodejs logo](https://drive.google.com/uc?export=view&id=13udtjL4Xz5YICREzII6hcHiv0xM8a_ds)
```
Comment (hectorkambow) [inline articles/history-of-nodejs/history-of-nodejs.md:24]: please include all images within the article itself - within the file folder - see image above as an example

File: articles/history-of-nodejs/history-of-nodejs.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+- **Features**
+    - Easy to learn
+    - Fast
+    - No buffering
+    - Consistent source code
+    - Massive library support
+
+### The Early Beginnings
+Believe it or not, Node.js is only twelve years old. In comparison, JavaScript is 26 years old and the Web is 32 years old.
+In 2009, [Ryan Dahl wrote Node.js](https://en.wikipedia.org/wiki/Node.js).  At first Node.js supported only Mac OS X and Mac OS X. Dahl led the development and maintenance and later it was [sponsored by Joyent](https://en.wikipedia.org/wiki/Node.js).
```
Comment (hectorkambow) [inline articles/history-of-nodejs/history-of-nodejs.md:44]: I think there may be a typo or duplicate where its says Mac OS X and Mac OS X?

File: articles/history-of-nodejs/history-of-nodejs.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+The limited possibilities of the most popular web server at the time ["Apache HTTP Server"](https://en.wikipedia.org/wiki/Apache_HTTP_Server) in 2009 was criticized by Dahl, because it had to handle a lot of connections concurrently (up to 10,000 and more) and when there was a blocked code in the entire process or an implied multiple execution stacks in cases of simultaneous connections, it would lead to issues, and this situation had to be resolved by creating code through [sequential programming](https://en.wikipedia.org/wiki/Sequential_algorithm).
+On November 8th 2009 at the inaugural European [JSConf the Node.js project was demonstrated by Dahl](https://en.wikipedia.org/wiki/Node.js). Node.js is a combination the V8 JavaScript chrome engine, a low-level I/O API and an event loop.
+
+### The Evolution
+![Nodejs image evolution](https://drive.google.com/uc?export=view&id=1e91n-R0-W4S6UOffTNTnBdqyJKNqzOXN)
+[source](https://nodejs.dev/learn/a-brief-history-of-nodejs)
```
Comment (hectorkambow) [inline articles/history-of-nodejs/history-of-nodejs.md:50]: Fantastic - thank you for including all image sources

File: articles/history-of-nodejs/history-of-nodejs.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+The file is saved on your computer: *C:\Users\YourName\firstfile.js*
+This code tells the computer to display "Hello World!" when anyone with a web browser tries to access your computer on port 8080.
+
+**Using Command Line Interface**
+To be able to access the Node.js files it must be initiated in the ["Command Line Interface" program of your computer.](https://www.w3schools.com/nodejs/nodejs_get_started.asp)
+Depending on your operating system, opening the command line interface on your computer would vary. If your are a Windows users, click on the start button and look for "Command Prompt", or you can simply write "cmd" in the search field for a quick search.
+
+After opening the 'cmd' go to the folder containing the file _"myfirstfile.js"_, the command line interface window should look something like this:
+
+`[C:\Users\YourName>](https://www.w3schools.com/nodejs/nodejs_get_started.asp)`
```
Comment (hectorkambow) [inline articles/history-of-nodejs/history-of-nodejs.md:134]: I added this in code blocks - 
It had a > in front of it - but in .md file anything after > doesn't show up (or is added as a note) 
Not sure if this was intended.

**Comment by hectorkambow** [comment]
@ninjaginja  👋  Good afternoon Molly - wanted to ask if you could give this History of Node.js article a tech review before we publish when you have a moment 🚀 
A couple in article images - ( i left these two out of all of them ) because they had text / info on them. 

Some minor code blocks. 
Thank you

**Comment by hectorkambow** [comment]
@Jethro-magaji  - Please go over (and pay close attention) to the _Formatting readability edits_ made to the final version to better incorporate those types of changes into future articles. 🚀 
Thank you

**Comment by Jethro-magaji** [comment]
Thanks, @hectorkambow I will consider edits next time.

**Pull request merged**